app/services/user_service.js

```diff
@@ -0,0 +1,18 @@
+const User = require('../models').user,
+  creationError = require('../errors').creationError,
+  bcrypt = require('bcrypt');
+
+const SALT_ROUNDS = 10;
+
+exports.createUser = user => {
```

Comment:
You could just write this function like this:
```
exports.createUser = user => {
  return bcrypt
    .hash(user.password, SALT_ROUNDS)
    .then(hash => {
      user.password = hash;
      return User.create(user);
    })
    .catch(err => {
      throw creationError;
    });
};
```
There is no need to add an additional `then` for the `User.create` and, even if you did, you could have just wrote `.then(User.create)` since you were taking the response from the previous promise as a parameter for the next one.
Also there is no need to use `Promise.reject` here, just throw the error.

Reply:
This one was a tricky one: 
1) I wasn't able to write `.then(User.create)` because the context mess up and I ended up getting an error
2) regarding using everything in the same `.then`, I thought that this way was clearer on what responsabilities each `then` has. If you think this is wrong or there are other reasons why it shouldn't be like this I am surely changing it
3) Mishu told me that throwing an error was way more inefficient that rejecting the promise, but if it is the way we are doing it I can change it.

Reply:
no but what Mishu meant was for example this
```
return new Promise((resolve, reject) => {
  // here you should call reject rather than throw an error
}
```

Reply:
regarding `I wasn't able to write .then(User.create) because the context mess up and I ended up getting an error` what error were you getting??

Reply:
It was trying to use the method `build` on undefined and in the code of the lib it was doing `this.build...`